Model/Sales/MagentoOrderCollection.php

```diff
@@ -222,10 +225,24 @@ public function setFulfilment()
         $orderLines      = new Collection();
 
         foreach ($this->getOrders() as $magentoOrder) {
+            $defaultOptions          = new DefaultOptions($magentoOrder, $this->helper);
+            $shipmentOptionsHelper   = new ShipmentOptions(
+                $defaultOptions,
+                $this->helper,
+                $magentoOrder,
+                $this->objectManager,
+                $this->options
+            );
             $myparcelDeliveryOptions = $magentoOrder['myparcel_delivery_options'];
-            $deliveryOptions         = json_decode($myparcelDeliveryOptions, true);
-            $deliveryOptionsAdapter  = DeliveryOptionsAdapterFactory::create((array) $deliveryOptions);
-            $this->order             = $magentoOrder;
+            $deliveryOptions         = json_decode($myparcelDeliveryOptions, true, 512, JSON_THROW_ON_ERROR);
```

Comment:
Dit is mogelijk gevaarlijk bij orders die geen delivery options hebben (dat kan als je een order kopieert of wanneer om wat voor reden dan ook, bv omdat eerst een andere verzendmethode was gekozen, de frontend geen delivery options heeft opgeslagen), ik zou eerder hier een foutafhandeling op zetten. Wordt dit specifiek getest?